packages/core/src/types-hoist/options.ts

```diff
@@ -347,15 +347,15 @@ export interface ClientOptions<TO extends BaseTransportOptions = BaseTransportOp
   beforeSend?: (event: ErrorEvent, hint: EventHint) => PromiseLike<ErrorEvent | null> | ErrorEvent | null;
 
   /**
-   * This function can be defined to modify or entirely drop a child span before it's sent.
+   * This function can be defined to modify a child span before it's sent.
    * Returning `null` will cause this span to be dropped.
    *
    * Note that this function is only called for child spans and not for the root span (formerly known as transaction).
    * If you want to modify or drop the root span, use {@link Options.beforeSendTransaction} instead.
    *
    * @param span The span generated by the SDK.
    *
-   * @returns A new span that will be sent or null if the span should not be sent.
+   * @returns A new or modified span that will be sent.
```

Comment:
```suggestion
   * @returns The modified span payload that will be sent.
```

maybe this is clearer?